File: UnitTests/CoffeeMaker/Python/coffee_maker.py
import string


class CoffeeMaker:
    coffeeSelect = None

    def __init__(self, coffee_select):
        self.CoffeeSelect = coffee_select

    def making_coffee(self) -> string:
        beans = self.CoffeeSelect.get_beans()
        if beans <= 0:
            return "No coffee available"
        if not self.CoffeeSelect.grind_beans():
            return "No ground coffee available";
        return self._brew(beans)

    def _brew(self, beans) -> string:
        brew = "Coffee has now been brewed with {} beans"
        return brew.format(beans);

File: UnitTests/CoffeeMaker/Java/src/main/java/CoffeeMaker.java
import Interfaces.ICoffeeMaker;
import Interfaces.ICoffeeSelect;
import jdk.jshell.spi.ExecutionControl;

public class CoffeeMaker implements ICoffeeMaker {

    private ICoffeeSelect coffeeSelect;

    public CoffeeMaker(ICoffeeSelect coffeeSelect)
    {
        this.coffeeSelect = coffeeSelect;
    }

    @Override
    public String MakingCoffee() throws ExecutionControl.NotImplementedException {
        if (coffeeSelect.GetBeans() <= 0)
        {
            return "No coffee available";
        }
        if (!this.coffeeSelect.GrindBeans())
        {
            return "No ground coffee available";
        }
        return this.brew();
    }

    private String brew()
    {
        return "Coffee has now been brewed";
    }
}
